src/Http/BaseApplication.php

```diff
@@ -110,6 +118,10 @@ public function __invoke(ServerRequestInterface $request, ResponseInterface $res
      */
     protected function getDispatcher()
     {
-        return new ActionDispatcher(null, null, DispatcherFactory::filters());
+        if ($this->dispatcher === null) {
```

Comment:
@markstory in that case it wouldn't if we wanted to support dispatcher filters while injecting a dispatcher on app creation.

I'll hold with this one and try a different middleware-based approach as discussed on slack :)